### PR TITLE
acceptance: deflake TestDockerCLI/test_server_restart.tcl

### LINKED ISSUE #15474: sql: DROP after restart hangs
The following tests appear to have failed:

[#237218](https://teamcity.cockroachdb.com/viewLog.html?buildId=237218):

```
--- FAIL: acceptance/TestDockerCLI (157.590s)
test_log_scope.go:79: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerCLI078706760
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestDockerCLI/test_server_restart.tcl (37.030s)
cli_test.go:73: non-zero exit code: 1

```

(dup of #15398)

### PR DESCRIPTION
**Pull request by petermattis**
Fixes #15474

### CODE REVIEW

**Comment by petermattis** [comment]
I can't actually reproduce a failure locally, so I'll run the acceptance tests multiple times on TeamCity before declaring that this fixes the problem.

**Comment by petermattis** [comment]
Also, perhaps `stop_server` should unconditionally do `$argv quit`. I'll make that change assuming TeamCity shows this works.

File: pkg/cli/interactive_tests/test_server_restart.tcl [hunk 1]
```diff
@@ -13,7 +13,11 @@ eexpect ":/# "
 send "$argv sql -e \"create database t; create table t.t (x INT); select * from t.t;\"\r"
 eexpect "(0 rows)"
 
-stop_server $argv
+# Send an interrupt to allow the server to gracefully shutdown. We
+# need a graceful shutdown in order for table leases to be released,
+# otherwise the DROP TABLE below will have to wait for the table lease
+# to expire (5 min).
+interrupt
```
Comment (knz) [inline pkg/cli/interactive_tests/test_server_restart.tcl]: `interrupt` sends an interrupt to the foreground process (the client), not the server.

You need to do `system "kill -INT `cat server_pid`"` to achieve this.

Comment (petermattis) [inline pkg/cli/interactive_tests/test_server_restart.tcl]: Heh, I just saw this. I'll just bite the bullet and change `stop_server`.

**Review by knz — APPROVED** [review]

**Comment by bdarnell** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15485#-:-KiqCnjI0Y29P8NmSo9Q:bnfp4nl)*

**Comment by knz** [comment]
Ok I am merging this so that a bunch of other deflake PRs can go through.

**Pull request merged**